packages/firestore/src/core/event_manager.ts

```diff
@@ -55,11 +56,34 @@ export class EventManager {
 
   private onlineState: OnlineState = OnlineState.Unknown;
 
-  constructor(private syncEngine: SyncEngine) {
+  constructor(private asyncQueue: AsyncQueue, private syncEngine: SyncEngine) {
     this.syncEngine.subscribe(
       this.onChange.bind(this),
       this.onError.bind(this)
     );
+    this.registerVisibilityListener();
+  }
+
+  private registerVisibilityListener(): boolean {
+    if (typeof window !== 'undefined') {
```

Comment:
I opted to put this into the EventManager since it is at least partially responsible for handling events from the user (well, only for 'listen' and 'unlisten') and I consider tab visibility user events. I agree that it is not necessarily the best fit, and since it introduces a dependency on the asyncQueue, I updated the PR to add a seperate class (platform_browser/window_event_listener).